tests/testthat/test-vctr.R

```diff
@@ -92,6 +92,19 @@ test_that("equality functions remapped", {
   expect_true(anyDuplicated(x))
 })
 
+test_that("is.na<-() and new_vctr() commute", {
+  x <- new_vctr(1:4)
+  y <- 1:4
+
+  expect_silent(is.na(x) <- c(FALSE, FALSE, TRUE, NA))
```

Comment:
I still don’t understand what you are trying to test here. I think it would make more sense to set a specific value to NA with is.na<- and then check that the value at that position equals the correct NA.

Reply:
Simplified the test, still testing the entire vector (because `is.na<-()` isn't supposed to touch other values).